src/s3/http.rs

```diff
@@ -379,80 +481,8 @@ impl BaseUrl {
     /// // Get base URL from IPv6 address
     /// let base_url = BaseUrl::from_string("[0:0:0:0:0:ffff:c0a8:7c3f]:9000".to_string()).unwrap();
     /// ```
+    #[deprecated(since = "0.1.0", note = "please use `from_str` instead")]
```

Comment:
We could remove this warning and remove `from_string()` as we haven't made any release yet.

Reply:
Done.